### DISCUSSION TITLE
Android audio backend

### DISCUSSION THREAD

**Discussion by provokateurin** [librespot-org/librespot#616] [Feature Requests]
I'm new to rust and this project, so please correct my mistakes:
I want to use librespot on Android and was already able to include it in my app and start playing, but it crashes while playing. My expectation was that it would crash, because there is no audio backend implementation for Android.
Is that right?

If so I would like to make a request for an Android audio backend.
I don't know what the requirements for a library that would work are, but figured out there is a library called `oboe` which is and Android audio library. Would that work?

As I need to learn Rust for my project (at least a bit) I might come up with a solution on my own (and will try to upstream it), but I wanted you to know that there is a need for an Android audio backend.

(I hope I haven't said anything really dumb.)

Thanks in advance and thank you all for such a great project! I really appreciate your effort.

EDIT: I also found this library which seems much more suited: https://github.com/AmpMe/opensles-sys

**Comment by mr-berndt** [comment]
From what I understand developers are on the long run actually trying to drop all audio outputs exept rodio. You might investigate if rodio works under android.

I got this from here:
https://github.com/librespot-org/librespot/issues/195

**Comment by Pajn** [comment]
rodio is using cpal, which seems to have recently gotten support for Android https://github.com/RustAudio/cpal/pull/468

**Comment by provokateurin** [comment]
Oh nice

**Comment by major-mayer** [comment]
So is there now a possibility to compile librespot with Android Audio backend support? 

If I understood it correctly, rodio should support oboe now,  but looking at the compile page [here ](https://github.com/librespot-org/librespot/wiki/Compiling) I can only see the WASAPI, Alsa and CoreAudio listed?

Did I miss something? I would love to build the Kodi librespot addon with Android support, so that I can use it on my Nvidia Shield.
